dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java

```diff
@@ -518,7 +518,7 @@ public TaskInstance getTaskInstance(int taskInstanceId) {
     private boolean processStateChangeHandler(StateEvent stateEvent) {
         try {
             logger.info("process:{} state {} change to {}", processInstance.getId(), processInstance.getState(), stateEvent.getExecutionStatus());
-            if (processComplementData()) {
+            if (processInstance.getState() != ExecutionStatus.READY_STOP && processComplementData()) {
```

Comment:
Is it better to move this state condition into `processComplementData()`?

Reply:
please review again.thanks.